Templates/Per Tenant Extension/.github/workflows/CreateApp.yaml

```diff
@@ -60,40 +61,21 @@ jobs:
         uses: microsoft/AL-Go-Actions/ReadSettings@main
         with:
           shell: powershell
-          parentTelemetryScopeJson: ${{ steps.init.outputs.telemetryScopeJson }}
           get: type
 
       - name: Read secrets
+        id: ReadSecrets
         uses: microsoft/AL-Go-Actions/ReadSecrets@main
-        if: github.event.inputs.useGhTokenWorkflow == 'true'
         with:
           shell: powershell
-          parentTelemetryScopeJson: ${{ steps.init.outputs.telemetryScopeJson }}
           gitHubSecrets: ${{ toJson(secrets) }}
-          getSecrets: 'ghTokenWorkflow'
-
-      - name: CalculateToken
-        id: CalculateToken
-        env:
-          useGhTokenWorkflow: ${{ github.event.inputs.useGhTokenWorkflow }}
-        run: |
-          $ghToken = '${{ secrets.GITHUB_TOKEN }}'
-          if ($env:useGhTokenWorkflow -eq 'true') {
-            $secrets = $env:Secrets | ConvertFrom-Json
-            if ($secrets.GHTOKENWORKFLOW) {
-              $ghToken = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($secrets.GHTOKENWORKFLOW))
-            }
-            else {
-              Write-Host "::Warning::GHTOKENWORKFLOW secret not found. Using GITHUB_TOKEN."
-            }
-          }
-          Add-Content -Encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "ghToken=$ghToken"
+          getSecrets: 'TokenForCommits'
 
       - name: Creating a new app
         uses: microsoft/AL-Go-Actions/CreateApp@main
         with:
           shell: powershell
-          token: ${{ steps.CalculateToken.outputs.ghToken }}
+          token: ${{ steps.ReadSecrets.outputs.TokenForCommits }}
```

Comment:
Good idea - will try

Reply:
The problem is really that we need to use GHTokenWorkflow coming back from ReadSecrets (as this might be from an Azure KeyVault) and it is base64 encoded in steps.ReadSecrets.outputs.GhTokenWorkflow - and Secrets.github_token is not.

Reply:
One way to do this would be to use ReadSecrets with the right name of the token - and then transfer secrets + the name of the secret to use:
![image](https://github.com/microsoft/AL-Go/assets/10775043/8e85b588-10de-4211-ba9d-c62d575f6aa4)

Reply:
I don't really like this construct - I like this better:
![image](https://github.com/microsoft/AL-Go/assets/10775043/120d5a6c-25ec-46f1-9610-41908fb49354)

which gets rid of the global ENV variable - and then the TokenForCommits is not base64 encoded, but still in the Secrets construct (we know that github tokens doesn't have any special characters).
we could also stick to the rule and base64 encode it - but then all actions would need to change as well to decode it.
or keep the extra output.

Reply:
Sorry for thinking out load here... - I think it makes sense to move the ENV to an input parameter - but I want to keep the extra output parameter for the unencoded TokenForCommits to use to centralize the decision inside the ReadSecrets for which secret to use based on parameters.